core/src/util/topologicalSort.h

```diff
@@ -0,0 +1,68 @@
+#pragma once
+
+#include <algorithm>
+#include <unordered_map>
+#include <utility>
+#include <vector>
+
+/**
+ * Given a list of edges describing a directed acyclic graph, returns a
+ * list of nodes sorted in topological order; that is, if there is a path
+ * from node 'a' to node 'b' in the graph, then 'a' will always precede
+ * 'b' in the sorted result.
+ */
+template<typename T>
+std::vector<T> topologicalSort(std::vector<std::pair<T, T>> edges) {

```

Comment:
Oh yeah, I probably should since we've done that with all our other utility classes. Will do!